packages/smart-wallet/src/walletFactory.js

```diff
@@ -36,9 +37,16 @@ export const publishDepositFacet = async (
   namesByAddressAdmin,
 ) => {
   const { nameHub, myAddressNameAdmin } = makeMyAddressNameAdminKit(address);
-  myAddressNameAdmin.update(WalletName.depositFacet, wallet.getDepositFacet());
+  myAddressNameAdmin.reserve(WalletName.depositFacet);
 
-  return E(namesByAddressAdmin).update(address, nameHub, myAddressNameAdmin);
+  // Don't clobber an existing myAddressNameAdmin or deposit facet.
```

Comment:
this will silently fail if `publishDepositFacet` is called a second time and `wallet.getDepositFacet()` changes.

This does seem like an improvement to make it immutable, but shouldn't we error if something tries to change it? At least log?

Reply:
More like it silently succeeds, yes?
This is saying "if there's already one there, leave it alone. Otherwise, use this one"

Reply:
Success or failure depends on the intent ;-)

My point is: if what's already there doesn't match, isn't that a surprise? In what cases would you expect the existing value to be different than the new one?

Reply:
> In what cases would you expect the existing value to be different than the new one?

In case an ag-solo already registered it.

> isn't that a surprise?

Not any longer, IIUC.

Reply:
I'll add documentation to this effect.